src/Data/List/Relation/Binary/Permutation/Propositional/Properties.agda

```diff
@@ -342,3 +342,21 @@ module _ {ℓ} {R : Rel A ℓ} (R? : Decidable R) where
     (x ∷ xs) ++ y ∷ ys       ≡˘⟨ Lₚ.++-assoc [ x ] xs (y ∷ ys) ⟩
     x ∷ xs ++ y ∷ ys         ∎
     where open PermutationReasoning
+
+------------------------------------------------------------------------
+-- foldr of Commutative Monoid
+
+module _ {c ℓ} (cmonoid : CommutativeMonoid c ℓ) where
+  open module CM = CommutativeMonoid cmonoid
+  import Relation.Binary.Reasoning.Setoid setoid as S
+  open import Algebra.Solver.CommutativeMonoid cmonoid
+
+  foldr-cmonoid : ∀ {xs ys} → xs ↭ ys → foldr _∙_ ε xs ≈ foldr _∙_ ε ys
```

Comment:
Can we make this name more standard by changing it to `foldr-commMonoid`?

Reply:
I renamed it to foldr-commMonoid.